kin-devplatform-sdk/src/main/java/kin/devplatform/accountmanager/AccountManagerImpl.java

```diff
@@ -254,9 +254,8 @@ public void onMigrationStart() {
 				@Override
 				public void onReady(IKinClient kinClient) {
 					Logger.log(new Log().priority(Log.DEBUG).withTag(TAG).text("onReady"));
-					blockchainSource.updateKinClientAndAccount(kinClient, accountIndex);
 					migrationManagerCallbacks.onReady(kinClient);
-					updateWalletAddress(accountIndex, callback);
+					updateWalletAddress(kinClient, accountIndex, callback);
```

Comment:
don't update here at all. it should be updated only after `updateWalletAddress` have succeeded.